### DISCUSSION TITLE
"How it Works" docs vs. CLI

### DISCUSSION THREAD

**Discussion by th0ger** [quarto-dev/quarto-cli#6580] [Q&A] [answered]
### Description

Quarto: [How it Works](https://quarto.org/docs/get-started/hello/vscode.html#how-it-works) - This seems clear when getting started.

My next questions were: What does these `.md` look like? What is the processing flow for (executed) `ipynb`?

In real life, the CLI says:
```bash
$ quarto render hello-quarto.qmd
Starting python3 kernel...Done
Executing 'hello-quarto.ipynb'
  Cell 1/3...Done
  Cell 2/3...Done
  Cell 3/3...Done

pandoc
  to: html
  output-file: hello-quarto.html
...
Output created: hello-quarto.html
```
and

```bash
$ quarto render hello-jupyter.ipynb
pandoc
  to: html
  output-file: hello-jupyter.html
...
Output created: hello-jupyter.html
```

So the actual flow look more like

UPDATE: Inserted `.md`:

```mermaid
flowchart LR
    ipynb[".ipynb<br/>(executed)"]
    qmd[".qmd"]
    iptmp[".ipynb<br/>(intermediate)"]
    md[".md<br/>(intermediate)"]
    pandoc(pandoc)
    ipynb -- convert --> md
    qmd -- quarto convert -->iptmp
    iptmp -- Jupyter exec-->iptmp
    iptmp -- convert --> md
    md --> pandoc
    pandoc --> .html
    pandoc --> .docx
    pandoc --> .pdf
```

What am I getting wrong? Is there a markdown at all?
(I dont use R.)

**Comment by mcanouil** [comment, marked as answer]
For short Pandoc reads from markdown.
There are `keep-*` options to keep the intermediate files, see <https://quarto.org/docs/computations/execution-options.html#intermediates>.

**Reply by th0ger** [reply]
Thanks, I have added the .md intermediate to the diagram.
